corehq/apps/app_manager/templates/app_manager/form_view_base.html

```diff
@@ -100,12 +100,12 @@
             {% if allow_form_filtering %}
             var putInRoot = {{ module.put_in_root|BOOL }};
             var isUserCaseInUse = {{ is_usercase_in_use|BOOL }};
+            var isModuleFilterEnabled = {{ is_module_filter_enabled|BOOL }};
             var formFilter = {{ form.form_filter|JSON }};
             var allOtherFormsRequireACase = {{ form.all_other_forms_require_a_case|BOOL }};
             var formFilterAllowed = ko.computed(function () {
-                return ((allOtherFormsRequireACase &&
-                         form_requires() === 'case') ||
-                        isUserCaseInUse) && !putInRoot;
+                return ((allOtherFormsRequireACase && form_requires() === 'case' && !putInRoot) ||
+                        isUserCaseInUse || isModuleFilterEnabled);
```

Comment:
This is not quite the same as just adding ` || isModuleFilterEnabled` to the original clause, because even if module filtering is not enabled, if the user case is available we don't care if the form is in the root or not.